### PR TITLE
Prevent unsanitized height from peer to cause deletion of txs in pool

### PR DESCRIPTION
**Pull request by lucafavatella**
~This PR is a draft patch proposed by @hanssv - I packaged it as a PR while I understand better its impact.~

Delivers https://www.pivotaltracker.com/n/projects/2124891/stories/158915027
* This PR does not prevent peer to delay GC, though that is considered smaller issue and tracked as https://www.pivotaltracker.com/story/show/158939771

### CODE REVIEW

File: apps/aecore/src/aec_tx_pool.erl [hunk 1]
```diff
@@ -463,13 +470,17 @@ pool_db_put(#state{ db = Db, gc_db = GCDb, gc_height = GCHeight }, Key, Tx, Even
...
+                            pool_db_raw_put(Db, GCDb, GCHeight, Key, Tx, Hash),
                             ok
                     end
             end
     end.
 
+pool_db_raw_put(Db, GCDb, GCHeight, Key, Tx, TxHash) ->
+    ets:insert(Db, {Key, Tx}),
+    enter_tx_gc(GCDb, TxHash, GCHeight + tx_ttl()),
+    ok.
```
Comment (eproxus) [inline apps/aecore/src/aec_tx_pool.erl]: No point in returning `ok` here since the return value is never matched.

**Comment by lucafavatella** [comment]
@sennui Can you please re-review/re-approve when you can, I performed minor amendment in distinct commit.

**Comment by lucafavatella** [comment]
I had to rebase for conflict on release notes.

**Comment by lucafavatella** [comment]
I see [test job failed](https://circleci.com/gh/aeternity/epoch/21273) for unrelated reason: I am restarting it.

```
%%% aehttp_integration_SUITE ==> all.internal_endpoints.list_oracle_queries: FAILED
%%% aehttp_integration_SUITE ==> {{badmatch,[]},
 [{aehttp_integration_SUITE,list_oracle_queries,1,
                            [{file,"/home/builder/epoch/_build/test/lib/aehttp/test/aehttp_integration_SUITE.erl"},
                             {line,2965}]},
  {test_server,ts_tc,3,[{file,"test_server.erl"},{line,1546}]},
  {test_server,run_test_case_eval1,6,[{file,"test_server.erl"},{line,1062}]},
  {test_server,run_test_case_eval,9,[{file,"test_server.erl"},{line,994}]}]}
```

**Pull request merged**